src/coreclr/src/vm/jitinterface.h

```diff
@@ -1479,6 +1516,12 @@ protected :
     ULONG32                 m_iNativeVarInfo;
     ICorDebugInfo::NativeVarInfo * m_pNativeVarInfo;
 
+#ifdef FEATURE_ON_STACK_REPLACEMENT
+    bool                    m_fOwnsPatchpointInfo;
```

Comment:
Is this needed because of `m_pPatchpointInfo` is overloaded to hold both old patchpoint info as well as newly generated patchpoint info? Would it be better to have two pointers to patchpoint instead?

Reply:
Yes, that's the reason -- could easily just be two separate pointer fields.